stock_move_auto_assign/models/product_product.py

```diff
@@ -38,4 +43,23 @@ def moves_auto_assign(self, locations):
         """
         self.ensure_one()
         moves = self.env["stock.move"].search(self._moves_auto_assign_domain(locations))
+        pickings = moves.picking_id
+        if not pickings:
+            return
+        try:
+            self.env.cr.execute(
```

Comment:
Is it really wanted ??? Shouldn't be a retryable job ?

Reply:
This will retry the job. The problem here is there's no conflict as if 2 jobs change the state of 2 different moves, none of them will update their common related picking, no concurrent write.

Each of the job will update their move, at the same time, so in both transactions, they'll have their respective move as "assigned" and the other one as "confirmed". So none will update the picking's state to "assigned".

I'm far to be convinced this is a good solution, but I have no other for now.

Reply:
Another option could be to use `move.picking_id.write({})`, that would update `write_date` on the stock.picking, so in case of concurrent move, one would retry, and we don't lock the stock.picking. But it means we have more code executed before the rollback happens.

Reply:
@jbaudoux do you have an opinion on the way to solve this issue?

Reply:
@guewen I like the explicit lock as you did.